multinet/db.py

```diff
@@ -307,21 +310,47 @@ def aql_query(
 def create_graph(
     workspace: str,
     graph: str,
-    node_tables: List[str],
     edge_table: str,
     arango: ArangoClient,
+    from_vertex_collections: Optional[Set[str]] = None,
+    to_vertex_collections: Optional[Set[str]] = None,
 ) -> bool:
     """Create a graph named `graph`, defined by`node_tables` and `edge_table`."""
     space = db(workspace, arango=arango)
     if space.has_graph(graph):
         return False
     else:
-        g = space.create_graph(graph)
-        g.create_edge_definition(
-            edge_collection=edge_table,
-            from_vertex_collections=node_tables,
-            to_vertex_collections=node_tables,
+        if not from_vertex_collections or not to_vertex_collections:
+            properties = get_edge_table_properties(workspace, edge_table)
+            _from_vertex_collections: Set[str] = properties["from_tables"]
+            _to_vertex_collections: Set[str] = properties["to_tables"]
+        else:
+            _from_vertex_collections = from_vertex_collections
+            _to_vertex_collections = to_vertex_collections
+
+        new_edge_def = {
+            "edge_collection": edge_table,
+            "from_vertex_collections": list(_from_vertex_collections),
+            "to_vertex_collections": list(_to_vertex_collections),
+        }
+
+        edge_defs = list(
+            chain.from_iterable([g["edge_definitions"] for g in space.graphs()])
         )
+        conflicts = [
+            e_def
+            for e_def in edge_defs
+            if e_def["edge_collection"] == edge_table and e_def != new_edge_def
+        ]
+
+        if conflicts:
+            # THIS SHOULDN'T HAPPEN
+            # TODO: Add error
+            print("CONFLICTING EDGE DEFINITIONS:", conflicts)
+            return False
```

Comment:
If this is truly an exceptional condition (meaning that something is seriously wrong with the setup, or something we never anticipated has happened), then this is a good place to raise a sort of fatal error. In an ordinary Python program, I would do something like `raise RuntimeError(f"Conflicting edge definitions: {conflicts}")`; in this Flask server, I'm not completely sure what to do, but it's probably a new class of exception mapping to a 500-level error.

Let me know if you want to discuss this further. This is a small but important change to make since we want to know immediately what happened if we ever run into this error condition.

Reply:
Following our discussion offline I've pushed these changes, let me know what you think.